driver-core/src/main/com/mongodb/MongoCredential.java

```diff
@@ -362,14 +362,22 @@ public MongoCredential withMechanism(final AuthenticationMechanism mechanism) {
      * @param source    the source of the user name, typically a database name
      * @param password  the password
      */
-    MongoCredential(@Nullable final AuthenticationMechanism mechanism, @Nullable final String userName, final String source,
-                    @Nullable final char[] password) {
+    MongoCredential(
+            @Nullable final AuthenticationMechanism mechanism,
+            @Nullable final String userName,
+            final String source,
+            @Nullable final char[] password) {
         this(mechanism, userName, source, password, Collections.emptyMap());
     }
 
-    MongoCredential(@Nullable final AuthenticationMechanism mechanism, @Nullable final String userName, final String source,
-                    @Nullable final char[] password, final Map<String, Object> mechanismProperties) {
-        if (mechanism != MONGODB_X509 && mechanism != MONGODB_AWS && userName == null) {
+    MongoCredential(
+            @Nullable final AuthenticationMechanism mechanism,
+            @Nullable final String userName,
+            final String source,
+            @Nullable final char[] password,
+            final Map<String, Object> mechanismProperties) {
+
+        if (userName == null && !Arrays.asList(MONGODB_X509, MONGODB_AWS).contains(mechanism)) {
```

Comment:
Is OIDC going to be added to this list?  Is that what motivated this change?

Reply:
Right.

(There was a greater upheaval of the validation logic here before, but I had to move the checks further down in the process.)